src/gui/sdl_gui.cpp

```diff
@@ -1766,6 +1770,81 @@ void GFX_Init()
 	        SDL_GetCurrentVideoDriver(),
 	        SDL_GetCurrentAudioDriver());
 
+#ifdef MACOSX
+	// On macOS, check for document drop events (double-click to open .dosbox packages)
+	// When a user double-clicks a .dosbox package in Finder, macOS delivers the
+	// file path via SDL_DROPFILE event rather than command-line arguments.
+	// We check for this immediately after SDL_Init but before the restart function
+	// is registered, so we implement the restart directly here.
+	SDL_Event event;
+	while (SDL_PollEvent(&event)) {
```

Comment:
Can we start polling immediately after startup? Isn't it possible for a race condition to happen here, don't we need to sleep "a bit" in the 50-200ms range perhaps before polling? Just wondering, I don't know 😄

Reply:
I vaguely remember I had to do that in one of my glfw apps on macOS that handle some window system events at startup, so yeah, probably it's worthwhile to add a bit of delay back.